### ISSUE TITLE
pytorch training getting NANs

### ISSUE THREAD

**Issue by zozobozo** [cogaplex-bts/bts#36]
Hi, 

Thanks a lot for releasing this code, and for taking the time to release the pytorch version too!

I tested your pre-trained model on kitti, as the README file suggests, and indeed got the expected results (rmse = 2.798). 
**However - training gets NANs after about 6.5K iterations**. (Before that, the loss got to ~1.0):

> [epoch][s/s_per_e/gs]: [1][1608/5790/7398], lr: 0.000097927423, loss: nan

1. **Could you perhaps verify the reason for getting NANs?**
2. **What should be the final loss at the final epoch?**

Thanks again for the great work!
Best,
Z.

Minor comments: 

1.  I'm using your recently-release pytorch version 

> "fa5face5eb76b16119613aebf9dc9965d6070e4d ; "Replacing bts_parameters with args"

2. The only difference from the setup you provide, is that i disable "multiprocessing_distributed", since it provides this error:

> OMP: Error #13: Assertion failure at z_Linux_util.cpp(2361).
> OMP: Hint Please submit a bug report with this message, compile and run commands used, and machine configuration info including native compiler and operating system versions. Faster response will be obtained by including all program sources. For information on submitting this issue, please see http://www.intel.com/software/products/support/.

**Comment by cogaplex-bts** [comment]
Thanks for your interest in our work. There are three factors mostly affect stability in training: learning rate, weight decay and eps in AdamW optimizer. First two parameters can be set with arguments --learning_rate and --weight_decay. If your batch size is lower than 16 which we used for the shared pretrained models, I recommend you to lower the learning rate. 1e-4 is a good choice. Lowering weight decay (to 1e-3 or 1e-4) can also help. Increasing eps is the final option but you can see a significant performance drop.
In our trainings, the final loss would be lower than 0.4.

**Comment by zozobozo** [comment]
Thanks for your super quick reply;
I re-ran the training with batch-size=24 and weight-decay of 1e-3 for 50 epochs; Now indeed the training converges, giving:

>      d1,      d2,      d3,  AbsRel,   SqRel,    RMSE, RMSElog,   SILog,   log10
>     0.954,   0.993,   0.998,   0.061,   0.254,   2.868,   0.098,   9.034,   0.027

As a side-note:  the default training parameters have batch_size=4, perhaps it's worth changing that?
Thanks!
Z.

**Comment by kHarshit** [comment]
Hi @cogaplex-bts ,

lowering the learning rate decreases the performance, and also forces to train longer. Can gradient clipping solve this issue? I'm unsure at what value should I clip gradients.
